src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,386 @@
+            useAndOperator = true;
+        }
+        sb.AppendLine("");
+
+        // Set environment for nuget.config
+        sb.AppendLine("");
+        sb.AppendLine("# Set custom nuget.config");
+        sb.AppendLine("ENV RestoreConfigFile=/NuGet.config");
+
+        // Find scenario-tests binary in context/scenario-tests
+        string? scenarioTestsBinary = Directory.GetFiles(Path.Combine(_contextDir, "scenario-tests"), "Microsoft.DotNet.ScenarioTests.SdkTemplateTests.dll", SearchOption.AllDirectories).FirstOrDefault();
+        if (scenarioTestsBinary == null)
+        {
+            throw new Exception("Scenario tests binary not found");
+        }
+        scenarioTestsBinary = scenarioTestsBinary.Replace(_contextDir, "").Replace("\\", "/");
+
+        // Set entry point
+        sb.AppendLine("");
+        sb.AppendLine($"ENTRYPOINT [ \"dotnet\", \"{scenarioTestsBinary}\", \"--dotnet-root\", \"/usr/share/dotnet\" ]");
+
+        string dockerfile = Path.Combine(_contextDir, Path.GetRandomFileName());
+        File.WriteAllText(dockerfile, sb.ToString());
+        return dockerfile;
+    }
+
+    private bool AnyTestFailures(string testResultSummary)
+    {
+        var parts = testResultSummary.Split(',')
+         .Select(part => part.Split(':').Select(p => p.Trim()).ToArray())
+         .Where(p => p.Length == 2)
+         .ToDictionary(p => p[0], p => int.Parse(p[1]));
+
+        return parts["Errors"] > 0 || parts["Failures"] > 0;
+    }
+
+    private void AddPackage(List<string> packageList, string prefix, DistroType distroType)
+    {
+        packageList.Add(Path.GetFileName(GetContentPackage(prefix, distroType)));
+    }
+
+    private string GetContentPackage(string prefix, DistroType distroType)
+    {
+        string matchPattern = DistroType.Deb == distroType ? "*.deb" : "*.rpm";
+        string[] rpmFiles = Directory.GetFiles(_contextDir, prefix + matchPattern, SearchOption.AllDirectories)
+            .Where(p => !Path.GetFileName(p).Contains("dotnet-runtime-deps-"))
+            .ToArray();
+        if (rpmFiles.Length == 0)
+        {
+            throw new Exception($"RPM package with prefix '{prefix}' not found");
+        }
+
+        return rpmFiles.OrderByDescending(f => f).First();
+    }
+
+    private string GetMatchingDepsPackage(string baseImage, DistroType distroType)
+    {
+        string matchPattern = "dotnet-runtime-deps-*.deb";
+        if (distroType == DistroType.Rpm)
+        {
+            string depsId = "fedora";
+            if (baseImage.Contains("fedora"))
+            {
+                depsId = "fedora";
+            }
+            else if (baseImage.Contains("centos"))
+            {
+                depsId = "centos";
+            }
+            else if (baseImage.Contains("rhel"))
+            {
+                depsId = "rhel";
+            }
+            else if (baseImage.Contains("opensuse"))
```

Comment:
I'm removing these deps packages, so we don't need these checks here.

```suggestion
            string? depsId = null;
            if (baseImage.Contains("opensuse"
```

Reply:
Thanks, I'll incorporate this in an offline change, along other proposed changes.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/054366b087810078c0e6e27f6f47edbbb5f765bd